sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -163,6 +164,63 @@ private[joins] object HashedRelation {
       new GeneralHashedRelation(hashTable)
     }
   }
+
+  /**
+   * Consume a `LocalNode` to create a `HashedRelation`. Because `LocalNode` is not an `Iterator`,
+   * we cannot use `apply` directly. Moreover, to avoid creating another layer of `Iterator`, we
+   * have to duplicate most codes of `apply` here.
```

Comment:
I actually think implementing the wrapper is better since it's not very complicated. Duplicate code in general is really bad. We can have something like the following in `LocalNode`.
```
def asIterator: Iterator[InternalRow] = new LocalNodeIterator(this)
```